distributed/_cancellation.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from distributed.metrics import time
+
+
+class Deadline:
```

Comment:
The module name is a bit confusing. Cancellation is a bit too generic for my taste. I'm not sure if this thing warrants a dedicated module

Reply:
> I'm not sure if this thing warrants a dedicated module

Agreed, as noted in the description I'm not sure where to best put this. I'm not a huge fan of cramming everything into `distributed.utils` either but that might be the best spot for now. Any suggestions?